WeartGloves/src/WeArtGlove.cpp

```diff
@@ -0,0 +1,1598 @@
+    }
+    jointListUrdfYarp = config.find("considered_joint_list").asList();
+
+    for (size_t i = 0; i < jointListUrdfYarp->size(); i++) {
+        consideredJointNameList.push_back(jointListUrdfYarp->get(i).asString());
+    }
+    yInfo() << LogPrefix << "considered urdf joint names: " << consideredJointNameList;
+
+    // Get Finger-Tip Frame Names
+    yarp::os::Bottle* fingerTipFrameListUrdfYarp;
+    if (!(config.check("finger_tip_frame_list") && config.find("finger_tip_frame_list").isList())) {
+        yError() << LogPrefix << "Unable to find finger tip frame list in the config file.";
+        return false;
+    }
+    fingerTipFrameListUrdfYarp = config.find("finger_tip_frame_list").asList();
+
+    for (size_t i = 0; i < fingerTipFrameListUrdfYarp->size(); i++) {
+        fingerTipFrameNameList.push_back(fingerTipFrameListUrdfYarp->get(i).asString());
+    }
+    yInfo() << LogPrefix << "finger-tip frame names: " << fingerTipFrameNameList;
+
+    // Get human hand link name
+    if (!(config.check("hand_link") && config.find("hand_link").isString())) {
+        yError() << LogPrefix << "Unable to find hand_link in the config file.";
+        return false;
+    }
+    humanHandLinkName = config.find("hand_link").asString();
+    yInfo() << LogPrefix << "human hand link name: " << humanHandLinkName;
+
+    // Get human hand finger names
+    yarp::os::Bottle* fingerListYarp;
+    if (!(config.check("human_finger_list") && config.find("human_finger_list").isList()))
+    {
+        yError() << LogPrefix << "Unable to find human_finger_list in the config file.";
+        return false;
+    }
+    fingerListYarp = config.find("human_finger_list").asList();
+
+    for (size_t i = 0; i < fingerListYarp->size(); i++)
+    {
+        humanFingerNames.push_back(fingerListYarp->get(i).asString());
+    }
+    yInfo() << LogPrefix << "human finger names: " << humanFingerNames;
+
+    // Fill the coupling Matrix
+    yarp::os::Bottle* closureToHumanJointYarp;
+    if (trackingAlgorithmType == "SCTA")
+    {
+        if (!(config.check("closure_to_human_joint_simple")) || !config.find("closure_to_human_joint_simple").isList())
+        {
+            yError() << LogPrefix << "couldn't find the proper coupling matrix";
+            return false;
+        }
+        closureToHumanJointYarp = config.find("closure_to_human_joint_simple").asList();
+
+        if (closureToHumanJointYarp->size() < 60)
+        {
+            yError() << LogPrefix << "size of the coupling matrix for simple tracking is:" << closureToHumanJointYarp->size() << "Required 60";
+            return false;
+        }
+        closureToHumanJoint.resize(20, N_THIMBLES);
+        for (int j = 0; j < closureToHumanJointYarp->size(); j++)
+        {
+            int col = j % N_THIMBLES;
+            int row = j / N_THIMBLES;
+            closureToHumanJoint(row, col) = closureToHumanJointYarp->get(j).asFloat64();
+        }
+    }
+    else if (trackingAlgorithmType == "ASTA")
```

Comment:
The code in this else branch is quite big. I would suggest splitting it in different sub methods

Reply:
What do you think of this: https://github.com/ami-iit/yarp-device-haptic-gloves/commit/9837be0bfb894f0dd5dad111191f61f2139b1d39

Reply:
Much better, thanks!